### PR TITLE
`CopySnippet`: Add web docs for `@isTruncated` (HDS-2444)

### PR DESCRIPTION
**Pull request by KristinLBradley**
### :pushpin: Summary
If merged, this PR will add documentation for `@isTruncated` to the `CopySnippet`

**Web docs:** https://hds-website-git-hds-2444-document-copysnippet-179bfd-hashicorp.vercel.app/components/copy/snippet?tab=code

### :link: External links

- Jira ticket: [HDS-2444](https://hashicorp.atlassian.net/browse/HDS-2444)
- Slack thread: https://hashicorp.slack.com/archives/C025N5V4PFZ/p1692735495406229

***

### 👀 Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed
- [ ] Confirm that A11y tests have been run locally for this component

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-2444]: https://hashicorp.atlassian.net/browse/HDS-2444?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: website/app/styles/pages/components/copy-snippet.scss [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+
+// COMPONENTS > COPY/COPY-SNIPPET
+
+.doc-page-constrain-copy-snippet-width {
```
Comment (didoo) [inline website/app/styles/pages/components/copy-snippet.scss:9]: can you use the existing pattern/naming conventions for page-specific styles?
https://github.com/hashicorp/design-system/blob/main/website/app/styles/pages/components/modal.scss#L8-L9

File: website/docs/components/copy/snippet/partials/code/how-to-use.md [hunk 2]
```diff
@@ -22,3 +22,18 @@ This indicates that the component should take up the full-width of the parent co
 ```handlebars
 <Hds::Copy::Snippet @textToCopy="e4rt-yg80-39kt" @isFullWidth={{true}} />
 ```
+
+### isTruncated
+When set to `true`, this constrains text to one-line and truncates it if it does not fit the available space.
+
+Care should be taken in choosing to use this feature as there are [accessibility concerns](/components/copy/snippet?tab=accessibility).
+
```
Comment (jorytindall) [inline website/docs/components/copy/snippet/partials/code/how-to-use.md:30]:
```suggestion
```

File: website/docs/components/copy/snippet/partials/code/how-to-use.md [hunk 3]
```diff
@@ -22,3 +22,18 @@ This indicates that the component should take up the full-width of the parent co
 ```handlebars
 <Hds::Copy::Snippet @textToCopy="e4rt-yg80-39kt" @isFullWidth={{true}} />
 ```
+
+### isTruncated
```
Comment (jorytindall) [inline website/docs/components/copy/snippet/partials/code/how-to-use.md:26]:
```suggestion
### isTruncated

```

File: website/docs/components/copy/snippet/partials/code/component-api.md [hunk 4]
```diff
@@ -13,6 +13,11 @@ This component uses [ember-cli-clipboard](https://github.com/jkusa/ember-cli-cli
   <C.Property @name="container" @type="string">
      Selector string or element object of containing element, typically used in conjunction with modals; set the focused element as the container value.
   </C.Property>
+  <C.Property @name="isTruncated" @type="boolean" @default="false">
+    Constrains text to one line and truncates it based on available width. Text will only be truncated if it does not fit within the available space within a page layout.
```
Comment (jorytindall) [inline website/docs/components/copy/snippet/partials/code/component-api.md:17]: [Suggestion] Could trim this sentence to not repeat "within" twice

```suggestion
    Constrains text to one line and truncates it based on available width. Text will only be truncated if it does not fit within the available space.
```

**Review by jorytindall — APPROVED** [review]
Left a couple of non-blocking comments

**Review by didoo — APPROVED** [review]
LGTM! 👍

**Pull request merged**